[regression] 1.11.0rc3 Fails to build > 20 packages from the openSUSE/GNOME stack

**Describe the bug**
to test meson release candidates, I have setup a test project building the openSUSE/GNOME stack against the new 1.11.0rc3 tarballs.

This results in [a lot of failures](https://build.opensuse.org/project/monitor/GNOME:Next?arch_x86_64=1&defaults=0&failed=1&repo_meson=1), which have not been observed before (including rc2!)


**To Reproduce**
Rebuild the GNOME stack using meson 1.11.0rc3

Packages observed failing to build include things like
* gtk4: Apparently an indirect error, as pango no longer declared the fribidi dependency in the .pc file (and our gtk4.spec does not explicitly require it) The difference in this case is in how pango.pc has been generated:
  ```
> diff -ur {standard,meson}/usr/lib64/pkgconfig/pango.pc
--- standard/usr/lib64/pkgconfig/pango.pc	2026-03-23 18:13:14.000000000 +0100
+++ meson/usr/lib64/pkgconfig/pango.pc	2026-03-23 18:13:14.000000000 +0100
@@ -5,8 +5,6 @@
 Name: Pango
 Description: Internationalized text handling
 Version: 1.57.1
-Requires: gobject-2.0 >=  2.82, harfbuzz >=  8.4.0
-Requires.private: glib-2.0 >=  2.82, gio-2.0 >=  2.82, fribidi >=  1.0.6, libthai >=  0.1.9, fontconfig >=  2.17.0, freetype2, xrender, xft >=  2.0.0, cairo >=  1.18.0
+Requires: gobject-2.0, harfbuzz
 Libs: -L${libdir} -lpango-1.0
-Libs.private: -lm
 Cflags: -I${includedir}/pango-1.0
```
 
* cairomm: fails to find ft2build.h, despite freetype2-devel being part of the buildroot; seems we miss the -I parameter to find the header now
  ```
[   13s] c++ -Icairomm/libcairomm-1.16.so.1.4.0.p -I. -I.. -I/usr/include/sigc++-3.0 -I/usr/lib64/sigc++-3.0/include -I/usr/include/cairo -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++17 -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fPIC -DCAIROMM_BUILD=1 -MD -MQ cairomm/libcairomm-1.16.so.1.4.0.p/context_surface_win32.cc.o -MF cairomm/libcairomm-1.16.so.1.4.0.p/context_surface_win32.cc.o.d -o cairomm/libcairomm-1.16.so.1.4.0.p/context_surface_win32.cc.o -c ../cairomm/context_surface_win32.cc
[   13s] In file included from ../cairomm/enums.h:24,
[   13s]                  from ../cairomm/surface.h:27,
[   13s]                  from ../cairomm/context_private.h:23,
[   13s]                  from ../cairomm/context_surface_win32.cc:20:
[   13s] /usr/include/cairo/cairo-ft.h:46:10: fatal error: ft2build.h: No such file or directory
[   13s]    46 | #include <ft2build.h>
[   13s]       |          ^~~~~~~~~~~~
[   13s] compilation terminated.
```

**Expected behavior**
Same packages fail/succeed to build as with earlier versions of meson (unless meson.build errors are being exposed)

**system parameters**
* Native build of meson for openSUSE/Tumbleweed - GNOME:Next addond project
* openSUSE Tuimbleweel
* Python 3.11
* Meson 1.11.0rc3 ([rc2 had some minor other issue](https://github.com/mesonbuild/meson/issues/15665), but that was fixed; the 4 packages identified then currently build with rc3)

--- a/mesonbuild/interpreter/interpreter.py
+++ b/mesonbuild/interpreter/interpreter.py
@@ -3371,7 +3371,11 @@ class Interpreter(InterpreterBase, HoldableObject):
         default_library = self.coredata.optstore.get_value_for(OptionKey('default_library', subproject=self.subproject))
         assert isinstance(default_library, str), 'for mypy'
         if default_library == 'shared':
-            return self.build_target(node, args, T.cast('kwtypes.SharedLibrary', kwargs), build.SharedLibrary)
+            # Intentionally pass shared_library_only=False so that dependencies
+            # end up in Requires.private.  Many libraries that refer to their
+            # dependencies' in their headers expect this so that those dependencies
+            # are added to the output of 'pkgconfig --cflags'.
+            return self.build_target(node, args, T.cast('kwtypes.SharedLibrary', kwargs), build.SharedLibrary, shared_library_only=False)
         elif default_library == 'static':
             return self.build_target(node, args, T.cast('kwtypes.StaticLibrary', kwargs), build.StaticLibrary)
         elif default_library == 'both':

